### ISSUE TITLE
No match for digital nes games

### ISSUE THREAD

**Issue by fekir** [libretro/libretro-database#1403]
For example, from below is not recognized by retroarch, even if it has an entry on no-intro.

**Comment by RobLoach** [comment]
:+1: Feel free to submit it to https://github.com/libretro/libretro-database/blob/master/metadat/homebrew/Nintendo%20-%20Nintendo%20Entertainment%20System.dat

**Comment by fekir** [comment]
@RobLoach what should I submit? The games are already upstream (nointro in this case)

**Comment by fekir** [comment]
There is a misunderstanding.

I have to modify the downloaded rom, otherwise retroarch does not recognize it.

Nes roms have an optional header.
No-intro overrides this header.

Thus the hashes do not match.

If I adapt the header, then the rom is recognized.

As no-intro also has headerless hashes, a more robust approach would be to calculate a headerless hash.

**Comment by RobLoach** [comment]
Correct. Some people have exported some set with the headers over at https://github.com/libretro/libretro-database/blob/master/dat/Nintendo%20-%20Nintendo%20Entertainment%20System.dat

Otherwise, Manual Scan is the best option.

**Comment by RobLoach** [comment]
The data provided from No-Intro lives at https://github.com/libretro/libretro-database/blob/master/metadat/no-intro/Nintendo%20-%20Nintendo%20Entertainment%20System.dat

RetroArch doesn't currently support skipping the NES header when scanning, hence our need to have a seperate one for with a common set of headered CRCs. The CRCs end up being different, so it's not duplicate.

**Comment by fekir** [comment]
> RetroArch doesn't currently support skipping the NES header when scanning, hence our need [...]

My proposal/feature request was to add that support ;)

> The CRCs end up being different, so it's not duplicate

Yes, but it is already possible to recognize those games, it's a duplicated effort.

But maybe this is the wrong repo for adding such feature?

**Comment by RobLoach** [comment]
Found out what's wrong...

When downloading the sets, we need to check toggle "Pirate, Homebrew, Aftermarket" :facepalm:

**Comment by RobLoach** [comment]
But even with that, I'm unsure the hashes are the same for the release vs what's in No-Intro.
